admin/admin-links.php

```diff
@@ -286,6 +286,106 @@ public function edit_term_translation_link( int $term_id, string $taxonomy, stri
 		return $this->edit_translation_link( $link, $language );
 	}
 
+	/**
+	 * Returns the html markup for a post edit link.
+	 *
+	 * @since 3.8
+	 *
+	 * @param WP_Post      $post     The post.
+	 * @param PLL_Language $language The language of the post.
+	 * @param string       $mode     Optional. How the link should be displayed: with a pen icon or a language's flag.
+	 *                               Possible values are `icon` and `flag`. Default is `icon`.
+	 * @return string
+	 *
+	 * @phpstan-param 'icon'|'flag' $mode
+	 */
+	public function get_edit_post_item_link_tag( WP_Post $post, PLL_Language $language, string $mode = 'icon' ): string {
+		$url = (string) get_edit_post_link( $post->ID );
+		return $this->get_edit_item_link( $url, $post->ID, $post->post_title, $language, $mode );
+	}
+
+	/**
+	 * Returns the html markup for a term edit link.
+	 *
+	 * @since 3.8
+	 *
+	 * @param WP_Term      $term      The term.
+	 * @param string       $taxonomy  Taxonomy name.
+	 * @param string       $post_type Post type name.
+	 * @param PLL_Language $language  The language of the term.
+	 * @param string       $mode      Optional. How the link should be displayed: with a pen icon or a language's flag.
+	 *                                Possible values are `icon` and `flag`. Default is `icon`.
+	 * @return string
+	 *
+	 * @phpstan-param 'icon'|'flag' $mode
+	 */
+	public function get_edit_term_item_link_tag( WP_Term $term, string $taxonomy, string $post_type, PLL_Language $language, string $mode = 'icon' ): string {
+		$url = (string) get_edit_term_link( $term->term_id, $taxonomy, $post_type );
+		return $this->get_edit_item_link( $url, $term->term_id, $term->name, $language, $mode );
+	}
+
+	/**
+	 * Returns a link to edit an item (or an icon/flag if the current user is not allowed to).
+	 *
+	 * @since 3.8
+	 *
+	 * @param string       $url       URL of the edition link.
+	 * @param int          $item_id   ID of the item.
+	 * @param string       $item_name Name of the item.
+	 * @param PLL_Language $language  Language of the item.
+	 * @param string       $mode      Optional. How the link should be displayed: with a pen icon or a language's flag.
+	 *                                Possible values are `icon` and `flag`. Default is `icon`.
+	 * @return string
+	 *
+	 * @phpstan-param 'icon'|'flag' $mode
+	 */
+	public function get_edit_item_link( string $url, int $item_id, string $item_name, PLL_Language $language, string $mode = 'icon' ): string {
```

Comment:
This could be merged with `edit_translation_link()` and be a protected if not a private method.